File: backend/src/models/topUpPoin.js
import { DataTypes } from "sequelize";
import db from "../config/database.js";
import User from "./user.js";

const TopUpPoin = db.define(
  "TopUpPoin",
  {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true,
    },
    topupId: {
      type: DataTypes.STRING,
      allowNull: false,
      unique: true,
    },
    purchaseId: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    invoiceNumber: {
      type: DataTypes.STRING,
      allowNull: false,
      unique: true,
    },
    userId: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },
    points: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },
    price: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },
    paymentMethod: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    status: {
      type: DataTypes.STRING,
      allowNull: false,
      defaultValue: "success",
    },
  },
  {
    timestamps: true,
    createdAt: "created_at",
    updatedAt: "updated_at",
    tableName: "topuppoin",
  }
);

TopUpPoin.belongsTo(User, { foreignKey: "userId" });
User.hasMany(TopUpPoin, { foreignKey: "userId" });

export default TopUpPoin;

File: backend-go/models/topup_poin.go
package models

import (
	"time"
)

type TopUpPoin struct {
	ID            uint      `gorm:"primaryKey;autoIncrement" json:"id"`
	TopupID       string    `gorm:"type:varchar(255);not null;uniqueIndex"`
	PurchaseID    string    `gorm:"type:varchar(255);not null"`
	InvoiceNumber string    `gorm:"type:varchar(255);not null;uniqueIndex"`
	UserID        uint      `gorm:"not null;index"`
	Points        int       `gorm:"not null"`
	Price         int       `gorm:"not null"`
	PaymentMethod string    `gorm:"type:varchar(255);not null"`
	Status        string    `gorm:"type:varchar(255);not null;default:'success'"`
	CreatedAt     time.Time `gorm:"autoCreateTime"`
	UpdatedAt     time.Time `gorm:"autoUpdateTime"`

	// Belongs To relationship with User
	User *User `gorm:"foreignKey:UserID"` // Pointer to avoid recursive issues
}

func (TopUpPoin) TableName() string {
	return "topuppoin"
}
